src/lib/core/data-table/data-source.ts

```diff
@@ -0,0 +1,6 @@
+import {Observable} from 'rxjs/Observable';
+import {CdkTableViewData} from './data-table';
+
+export abstract class DataSource<T> {
+  abstract connectTable(viewChanged: Observable<CdkTableViewData>): Observable<T[]>;
```

Comment:
This should be something like 
```ts
abstract connect(collectionViewer: CollectionViewer): Observable<T[]>;
```
Where `CollectionViewer` is an interface that both the data table and the tree would implement that exposes the necessary stream (and potentially other information in the future)